### PR TITLE
Fix user creation with empty password

### PR DESCRIPTION
**Pull request by cri-epita**
### What does this PR do?

Fix user creation with empty password

### What issues does this PR fix or reference?

Emptying the password was only done if the user already existed, not
when creating.

### Tests written?

No

### CODE REVIEW

**Review by twangboy — APPROVED** [review]
Since it's behind a gate for Windows I don't think it will affect Windows...

File: salt/states/user.py [hunk 1]
```diff
@@ -676,6 +676,14 @@ def present(name,
                                          ' {1}'.format(name, 'XXX-REDACTED-XXX')
                         ret['result'] = False
                     ret['changes']['password'] = 'XXX-REDACTED-XXX'
+                if empty_password:
+                    __salt__['shadow.del_password'](name)
```
Comment (cachedout) [inline salt/states/user.py]: Isn't this already happening up on L457?

Comment (cri-epita) [inline salt/states/user.py]: Not really.

If we are in the case of creating a user, L457 will try to remove the password of a non-existent user. If the user already exists, the password will be removed, without checking that it is necessary.

I modified the commit to verify that it is necessary and handle both cases:
 * Creating a user
 * User already present

There is now the password removing information in the changes.

Comment (cachedout) [inline salt/states/user.py]: Makes sense. Thanks for the clarification!

**Pull request merged**